pkg/types/alibabacloud/machinepool.go

```diff
@@ -29,18 +33,10 @@ type MachinePool struct {
 	// +optional
 	InstanceType string `json:"instanceType,omitempty"`
 
-	// SystemDiskCategory defines the category of the system disk.
-	//
-	// +kubebuilder:validation:Type=DiskCategory
-	// +optional
-	SystemDiskCategory DiskCategory `json:"systemDiskCategory,omitempty"`
-
-	// SystemDiskSize defines the size of the system disk in gibibytes (GiB).
+	// SystemDiskInfo defines the category of the system disk.
 	//
-	// +kubebuilder:validation:Type=integer
-	// +kubebuilder:validation:Minimum=120
 	// +optional
-	SystemDiskSize int `json:"systemDiskSize,omitempty"`
+	SystemDisk machinev1.SystemDiskProperties `json:"systemDisk,omitempty"`
```

Comment:
Is this change intentional? There is no mention of it in the commit description. Generally, it is discouraged to use a type defined elsewhere in the install config. Doing so makes it much more difficult to ensure the API stability of the install config.

Reply:
It was intentional to align with the new API types. I have gone back and replaced it with what was previously there from your comments.